server/src/main/java/org/opensearch/search/builder/SearchSourceBuilder.java

```diff
@@ -302,6 +305,9 @@ public SearchSourceBuilder(StreamInput in) throws IOException {
         if (in.getVersion().onOrAfter(Version.V_2_18_0)) {
             searchPipeline = in.readOptionalString();
         }
+        if (in.getVersion().onOrAfter(Version.CURRENT)) {
```

Comment:
I think you need to have exact version here, CURRENT will change with every next release

Reply:
I initially used the exact version here, but it caused failures in the BWC integration tests. @dbwiddis  suggested switching to CURRENT for now, with the plan to update the version in a separate PR when preparing for the release.